assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js

```diff
@@ -89,7 +61,7 @@ export default function WidgetAreaRenderer( { slug, totalAreas } ) {
 	} );
 
 	// Render all widgets.
-	const widgetsOutput = activeWidgets.map( ( widget, i ) => (
+	const widgetsOutput = widgets.map( ( widget, i ) => (
```

Comment:
Passing `widgets` here makes sense given the new approach, but we'll also need to pass `widgets` in the two utility function calls above, instead of passing `activeWidgets`. Otherwise the layout will be messed up, since the utility functions work based on indexes, and those are of course different between `widgets` and `activeWidgets`.

We'll need to consistently act on all widgets now (i.e. the `widgets` variable), and to do that we'll need to slightly expand the `getWidgetLayout` and `combineWidgets` utility functions to also support inactive widgets:
* The function parameters need to be renamed from `activeWidgets` to `widgets` to clarify intent.
* `getWidgetLayout` also needs to receive `widgetStates` to be able to check which widgets are inactive. For those, it should simply set the `classNames` entry to `null`, the `columnWidths` entry to `0`, and the `rowIndexes` to the current `rowIndex`. The other logic should be entirely bypassed for them, since it's only relevant for active widgets.
* `combineWidgets` should be pretty much already compatible, since it already has a clause that will set the `HIDDEN_CLASS` for any widgets that have their `classNames` entry set to `null`. I think here we only need to update the parameter name and its documentation, as mentioned above.
* We should add Jest test coverage for these new pieces of logic.

Reply:
Updated.